app/lib/features/activities/pages/activities_page.dart

```diff
@@ -53,15 +54,60 @@ class ActivitiesPage extends ConsumerWidget {
 
   Widget buildActivityBody(
     BuildContext context,
+    WidgetRef ref,
     List<Widget> sectionWidgetList,
   ) {
+    final lang = L10n.of(context);
     final isActivityEmpty = sectionWidgetList.isEmpty;
     if (isActivityEmpty) return buildEmptyStateWidget(context);
 
-    return SingleChildScrollView(
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: sectionWidgetList,
+    final hasMoreActivities = ref.watch(hasMoreActivitiesProvider);
+    final isLoadingMore = ref.watch(isLoadingMoreActivitiesProvider);
+    final loadMoreActivities = ref.watch(loadMoreActivitiesProvider);
+
+    return NotificationListener<ScrollNotification>(
+      onNotification: (ScrollNotification scrollInfo) {
+
+        if (scrollInfo is ScrollUpdateNotification) {
+          final pixels = scrollInfo.metrics.pixels;
+          final maxExtent = scrollInfo.metrics.maxScrollExtent;
+          final progress = maxExtent > 0 ? pixels / maxExtent : 0;
+          
+          // Check if user has scrolled to near the bottom (90% of the way)
+          // and ensure we have more data to load and aren't already loading
+          if (progress >= 0.9 &&
+              hasMoreActivities && 
+              !isLoadingMore &&
+              maxExtent > 0) {
```

Comment:
isn't progress already checking the case of `maxExtent` ? that seems unnecessary to check again ...